modules/carousel/jetpack-carousel.js

```diff
@@ -1726,13 +1730,13 @@ jQuery( document ).ready( function( $ ) {
 	// handle lightbox (single image gallery) for images linking to 'Attachment Page'
 	if ( 1 === Number( jetpackCarouselStrings.single_image_gallery ) ) {
 		processSingleImageGallery();
-		$( document.body ).on( 'post-load', function() {
+		document.body.addEventListener( 'post-load', function() {
```

Comment:
@gravityrail jQuery Event system [is not compatible](https://bugs.jquery.com/ticket/11047) with native events (also https://github.com/jquery/jquery/issues/2476, https://github.com/jquery/jquery/issues/3347).

As such, it doesn't allow us to just replace `.on` constructs with `.addEventListener`, because `addEventListener` is not going to respond to events emitted by `$( ... ).trigger`. To make things even more complicated, we can't simply replace `$( ... ).trigger` with custom events in cases when custom data is passed with the event, because jQuery injects the data as a parameter to the event handler, whereas in native events, data is passed in as `e.detail`.

In this particular case, we should listen to (the custom native) `is.post-load` event, because the `post-load` event [is triggered by and for jQuery](https://github.com/Automattic/jetpack/blob/82db44a44297c6cd3d49dad49334d21038ad1a2b/modules/infinite-scroll/infinity.js#L163-L166) only.

We also need to consider any 3rd party code using these events. I believe the safest way forward is to define _new_ custom native events alongside the existing jQuery events and to make sure both are triggered - at least for some grace period. Perhaps with a deprecation notice.

Reply:
Those are great points (I learned something for sure!) - but if we assume we _don't_ want to support third parties listening for those events, and we change both the emitter _and_ the listeners at the same time, is it safe to transition to native events?

Reply:
> if we assume we don't want to support third parties listening for those events, and we change both the emitter and the listeners at the same time, is it safe to transition to native events?

Yes, in that case it's safe to switch to the native event layer. 👍

It can just be a bit clumsy to selectively unbind native event handlers. In jQuery events can be namespaced, which makes the process trivial. In vanilla JS we need to retain the reference to the event listener.